File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/model/AudioMediaResource.java
package com.sdkwork.backend.model;

public class AudioMediaResource {
    private Integer id;
    private String uuid;
    private String url;
    private String bytes;
    private Object localFile;
    private String base64;
    private String type;
    private String mimeType;
    private Integer size;
    private String name;
    private String extension;
    private TagsContent tags;
    private Map<String, Object> metadata;
    private String prompt;
    private String format;
    private Integer bitRate;
    private Integer sampleRate;
    private Integer channels;
    private Integer duration;

    public Integer getId() {
        return this.id;
    }
    
    public void setId(Integer id) {
        this.id = id;
    }

    public String getUuid() {
        return this.uuid;
    }
    
    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getUrl() {
        return this.url;
    }
    
    public void setUrl(String url) {
        this.url = url;
    }

    public String getBytes() {
        return this.bytes;
    }
    
    public void setBytes(String bytes) {
        this.bytes = bytes;
    }

    public Object getLocalFile() {
        return this.localFile;
    }
    
    public void setLocalFile(Object localFile) {
        this.localFile = localFile;
    }

    public String getBase64() {
        return this.base64;
    }
    
    public void setBase64(String base64) {
        this.base64 = base64;
    }

    public String getType() {
        return this.type;
    }
    
    public void setType(String type) {
        this.type = type;
    }

    public String getMimeType() {
        return this.mimeType;
    }
    
    public void setMimeType(String mimeType) {
        this.mimeType = mimeType;
    }

    public Integer getSize() {
        return this.size;
    }
    
    public void setSize(Integer size) {
        this.size = size;
    }

    public String getName() {
        return this.name;
    }
    
    public void setName(String name) {
        this.name = name;
    }

    public String getExtension() {
        return this.extension;
    }
    
    public void setExtension(String extension) {
        this.extension = extension;
    }

    public TagsContent getTags() {
        return this.tags;
    }
    
    public void setTags(TagsContent tags) {
        this.tags = tags;
    }

    public Map<String, Object> getMetadata() {
        return this.metadata;
    }
    
    public void setMetadata(Map<String, Object> metadata) {
        this.metadata = metadata;
    }

    public String getPrompt() {
        return this.prompt;
    }
    
    public void setPrompt(String prompt) {
        this.prompt = prompt;
    }

    public String getFormat() {
        return this.format;
    }
    
    public void setFormat(String format) {
        this.format = format;
    }

    public Integer getBitRate() {
        return this.bitRate;
    }
    
    public void setBitRate(Integer bitRate) {
        this.bitRate = bitRate;
    }

    public Integer getSampleRate() {
        return this.sampleRate;
    }
    
    public void setSampleRate(Integer sampleRate) {
        this.sampleRate = sampleRate;
    }

    public Integer getChannels() {
        return this.channels;
    }
    
    public void setChannels(Integer channels) {
        this.channels = channels;
    }

    public Integer getDuration() {
        return this.duration;
    }
    
    public void setDuration(Integer duration) {
        this.duration = duration;
    }
}

File: sdkwork-backend-sdk-typescript/src/types/audio-media-resource.ts
import type { TagsContent } from './tags-content';

/** 音频媒体资源 */
export interface AudioMediaResource {
  /** 资源ID */
  id?: number;
  /** 资源UUID */
  uuid?: string;
  /** 资源URL */
  url?: string;
  /** 资源字节数据 */
  bytes?: string;
  /** 本地文件 */
  localFile?: unknown;
  /** 资源Base64编码 */
  base64?: string;
  /** 资源类型 */
  type?: 'IMAGE' | 'VIDEO' | 'AUDIO' | 'DOCUMENT' | 'FILE' | 'MUSIC' | 'CHARACTER' | 'MODEL_3D' | 'PPT' | 'CODE';
  /** 资源MIME类型 */
  mimeType?: string;
  /** 资源大小(字节) */
  size?: number;
  /** 资源名称 */
  name?: string;
  /** 资源扩展名 */
  extension?: string;
  /** 资源标签 */
  tags?: TagsContent;
  /** 资源元数据 */
  metadata?: Record<string, unknown>;
  /** AI生成提示词 */
  prompt?: string;
  /** 音频格式 */
  format?: 'wav' | 'mp3' | 'aac' | 'flac' | 'ogg' | 'pcm' | 'aiff' | 'au' | 'opus';
  /** 比特率 */
  bitRate?: number;
  /** 采样率 */
  sampleRate?: number;
  /** 声道数 */
  channels?: number;
  /** 音频时长(秒) */
  duration?: number;
}
